src/components/split-button/split-button.component.js

```diff
@@ -0,0 +1,263 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Icon from '../icon/icon';
+import Button from '../button';
+import StyledSplitButtonContainer, { StyledToggleButton } from './split-button.style';
+import StyledSplitButtonChildrenContainer from './split-button-children.style';
+import { validProps } from '../../utils/ether/ether';
+import OptionsHelper from '../../utils/helpers/options-helper';
+import Events from '../../utils/helpers/events';
+
+class SplitButton extends React.Component {
+  static propTypes = {
+    /**
+     * Customizes the appearance, can be set to 'primary' or 'secondary'.
+     */
+    as: PropTypes.string,
+
+    /**
+     * A custom value for the data-element attribute
+     */
+    'data-element': PropTypes.string,
+
+    /**
+     * A custom value for the data-element attribute
+     */
+    'data-role': PropTypes.string,
+
+    /**
+     * The additional button to display.
+     */
+    children: PropTypes.node.isRequired,
+
+    /**
+     * Custom className
+     */
+    className: PropTypes.string,
+
+    /**
+     * Gives the button a disabled state.
+     */
+    disabled: PropTypes.bool,
+
+    // keyNavigation: PropTypes.bool,
+
+    /**
+     * The size of the buttons in the SplitButton.
+     */
+    size: PropTypes.oneOf(OptionsHelper.sizesRestricted),
+
+    /**
+     * The text to be displayed in the SplitButton.
+     */
+    text: PropTypes.string.isRequired
+  }
+
+  static defaultProps = {
+    as: 'secondary',
+    disabled: false,
+    size: 'medium'
+  }
+
+  static safeProps = ['as', 'disabled', 'size']
+
+  constructor(props) {
+    super(props);
+    this.componentTags = this.componentTags.bind(this);
+    this.additionalButtons = [];
+    this.splitButtons = [];
+  }
+
+  state = {
+    /**
+     * Tracks whether the additional buttons should be visible.
+     */
+    showAdditionalButtons: false,
+    selectedIndex: -1, // defaults to nothing being highlighted
+    isClassic: null
```

Comment:
can't we use withTheme here (https://www.styled-components.com/docs/api#withtheme)?